### ISSUE TITLE
[$250] IOU - System message removed the receipt appears outside of transaction thread

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#45602]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.7-3
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Email or phone of affected tester (no customers):** gocemate+a668@gmail.com
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. Go to any chat> Submit manual expense with receipt
2. Open the receipt> 3-Dot menu> Delete receipt
3. Submit another manual expense with receipt to same user
4. Open the receipt> 3-Dot menu> Delete receipt

## Expected Result:
System message "removed the receipt" should be inside thread transaction and appear in real time

## Actual Result:
System message "removed the receipt" is outside thread transaction

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/user-attachments/assets/f660e276-7b62-4eda-8c14-7dd461b78e04

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~017ca180fde4a68a29</li>
        <li>Upwork Job ID: 1814004987221632137</li>
        <li>Last Price Increase: 2024-07-18</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @akinwale</details>

**Comment by lanitochka17** [comment]
@garrettmknight  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by rafaykr** [comment]
Dear Expensify Development Team,

I'm writing in response to your call for React Native developers to assist with your front-end unification project. After reviewing the GitHub issue #45602, I've identified a critical user experience problem involving the misplacement of system messages when deleting receipts. This issue not only affects user comprehension but also impacts the overall reliability of your expense management tool.

As an experienced developer specializing in cross-platform solutions, I understand the challenges you're facing in maintaining a secure system while innovating in the competitive finance sector. The need for a unified front-end that ensures consistency across all platforms resonates with me, as I've helped several companies overcome similar hurdles.

I propose the following solution to address the issue:

1. Locate the component responsible for rendering the transaction thread and system messages.
2. Modify the receipt deletion logic to ensure the system message is created and inserted into the correct position within the transaction thread data structure.
3. Update the UI rendering logic to display the "removed the receipt" message within the thread transaction, rather than outside it.
4. Implement real-time updates using React Native's state management (e.g., useState, useReducer) or a state management library like Redux to ensure immediate reflection of changes in the UI.
5. Add a unique identifier to each system message to prevent duplication and ensure proper placement.
6. Implement error handling to manage cases where the message insertion fails.
7. Add unit and integration tests to verify the correct placement and real-time appearance of the system message.

Implementation steps will include:

1. Identifying the relevant components and data structures in the codebase.
2. Modifying the receipt deletion function to create and insert the system message into the correct position in the thread data structure.
3. Updating the thread rendering component to display system messages inline with other transaction items.
4. Implementing a real-time update mechanism (e.g., websockets or polling) to push updates to the UI immediately.
5. Adding error handling and logging for troubleshooting.
6. Writing and running tests to ensure the solution works as expected across different scenarios.
7. Documenting the changes and updating any relevant user guides or developer documentation.

This solution will:
- Correctly place the "removed the receipt" system message within the thread transaction.
- Ensure real-time appearance of the message for immediate user feedback.
- Maintain consistency across all receipt deletion actions.

By implementing this fix, we'll enhance the user experience, reduce potential errors in expense tracking, and reinforce Expensify's reputation for reliability and innovation in the fintech space.

I'm prepared to submit a detailed technical proposal on the GitHub issue, outlining the exact changes needed to resolve this problem efficiently and without causing regressions. My commitment to delivering high-quality, scalable solutions aligns perfectly with Expensify's standards.

I look forward to the opportunity to contribute to Expensify's continuous improvement and help maintain your position as the leading expense management tool. Let's collaborate to create a seamless, user-friendly experience that sets Expensify apart in the fintech space.

Best regards,
Muhammad Rafay Khan

**Comment by garrettmknight** [comment]
@rafaykr thanks for your interest. Can you repost your proposal using the process explained in [our guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md)?

**Comment by rafaykr** [comment]
> @rafaykr thanks for your interest. Can you repost your proposal using the process explained in [our guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md)?
contributor details
expensify account email:kamham56@gmail.com 
upwork account : https://www.upwork.com/freelancers/~01510f6bfd2680dd92

Dear Expensify Development Team,

I'm writing in response to your call for React Native developers to assist with your front-end unification project. After reviewing the GitHub issue https://github.com/Expensify/App/issues/45602, I've identified a critical user experience problem involving the misplacement of system messages when deleting receipts. This issue not only affects user comprehension but also impacts the overall reliability of your expense management tool.

As an experienced developer specializing in cross-platform solutions, I understand the challenges you're facing in maintaining a secure system while innovating in the competitive finance sector. The need for a unified front-end that ensures consistency across all platforms resonates with me, as I've helped several companies overcome similar hurdles.

I propose the following solution to address the issue:

Locate the component responsible for rendering the transaction thread and system messages.
Modify the receipt deletion logic to ensure the system message is created and inserted into the correct position within the transaction thread data structure.
Update the UI rendering logic to display the "removed the receipt" message within the thread transaction, rather than outside it.
Implement real-time updates using React Native's state management (e.g., useState, useReducer) or a state management library like Redux to ensure immediate reflection of changes in the UI.
Add a unique identifier to each system message to prevent duplication and ensure proper placement.
Implement error handling to manage cases where the message insertion fails.
Add unit and integration tests to verify the correct placement and real-time appearance of the system message.
Implementation steps will include:

Identifying the relevant components and data structures in the codebase.
Modifying the receipt deletion function to create and insert the system message into the correct position in the thread data structure.
Updating the thread rendering component to display system messages inline with other transaction items.
Implementing a real-time update mechanism (e.g., websockets or polling) to push updates to the UI immediately.
Adding error handling and logging for troubleshooting.
Writing and running tests to ensure the solution works as expected across different scenarios.
Documenting the changes and updating any relevant user guides or developer documentation.
This solution will:

Correctly place the "removed the receipt" system message within the thread transaction.
Ensure real-time appearance of the message for immediate user feedback.
Maintain consistency across all receipt deletion actions.
By implementing this fix, we'll enhance the user experience, reduce potential errors in expense tracking, and reinforce Expensify's reputation for reliability and innovation in the fintech space.

I'm prepared to submit a detailed technical proposal on the GitHub issue, outlining the exact changes needed to resolve this problem efficiently and without causing regressions. My commitment to delivering high-quality, scalable solutions aligns perfectly with Expensify's standards.

I look forward to the opportunity to contribute to Expensify's continuous improvement and help maintain your position as the leading expense management tool. Let's collaborate to create a seamless, user-friendly experience that sets Expensify apart in the fintech space.

Best regards,
Muhammad Rafay Khan
